Sources/Turf/Geometries/LineString.swift

```diff
@@ -215,4 +215,114 @@ extension LineString {
         
         return closestCoordinate
     }
+
+    private func getSquareDistance(coordinate1: CLLocationCoordinate2D, coordinate2: CLLocationCoordinate2D) -> Double {
+        let dx = coordinate1.longitude - coordinate2.longitude
+        let dy = coordinate1.latitude - coordinate2.latitude
+        return dx * dx + dy * dy
+  }
+
+    private func simplifyRadialDistance(_ coordinates: [CLLocationCoordinate2D], tolerance: Double) -> [CLLocationCoordinate2D] {
+        guard coordinates.count > 2 else { return coordinates }
+
+        var prevCoordinate = coordinates[0]
+        var newCoordinates = [prevCoordinate]
+        var coordinate = coordinates[1]
+
+        for index in 1 ..< coordinates.count {
+            coordinate = coordinates[index]
+
+            if getSquareDistance(coordinate1: coordinate, coordinate2: prevCoordinate) > tolerance {
+                newCoordinates.append(coordinate)
+                prevCoordinate = coordinate
+            }
+        }
+
+        if prevCoordinate != coordinate {
+            newCoordinates.append(coordinate)
+        }
+
+        return newCoordinates
+    }
+
+    private func getSquareSegmentDistance(_ coordinate: CLLocationCoordinate2D, segmentStart: CLLocationCoordinate2D, segmentEnd: CLLocationCoordinate2D) -> CLLocationDistance {
+
+        var x = segmentStart.latitude
+        var y = segmentStart.longitude
+        var dx = segmentEnd.latitude - x
+        var dy = segmentEnd.longitude - y
+
+        if dx != 0 || dy != 0 {
+            let t = ((segmentStart.latitude - x) * dx + (coordinate.longitude - y) * dy) / (dx * dx + dy * dy)
+            if t > 1 {
+                x = segmentEnd.latitude
+                y = segmentEnd.longitude
+            } else if t > 0 {
+                x += dx * t
+                y += dy * t
+            }
+        }
+
+        dx = coordinate.latitude - x
+        dy = coordinate.longitude - y
+
+        return dx * dx + dy * dy
+    }
+
+    private func simplifyDouglasPeuckerStep(_ coordinates: [CLLocationCoordinate2D], first: Int, last: Int, tolerance: Double, simplified: inout [CLLocationCoordinate2D]) {
+
+        var maxSquareDistance = tolerance
+        var index = 0
+
+        for i in first + 1 ..< last {
```

Comment:
This algorithm was ported somewhat literally from JavaScript, so it doesn’t make use of functional programming that would insulate us from potential off-by-one errors and the like. To give us more confidence, can you port some of the unit tests from JavaScript to Swift?

Reply:
I brought over the test for LineString from the JS version.
I am curious what changes you'd suggest in the code though.

Reply:
Thanks for bringing over the tests. In my opinion, it’s OK to leave it as a straight port of the JavaScript version if we haven’t gotten to understand the code that deeply. The tests at least give us confidence that we didn’t make an off-by-one error when porting the functionality, especially since the range operators don’t have an equivalent in JavaScript. At some point, we might want to revisit this code to make it more idiomatic and functional, with a possible efficiency benefit. Then the unit tests will also protect us against regressions.